### PR TITLE
Migrate the create a call functionality to call the calling library

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
## Purpose

* Implement a share mechanism for created group ID calls
* Migrate the create a call functionality to call the calling library

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
* Ensure you have set up AAD
* Go to invite screen and observe new design and share functionality

## What to Check
Verify that the following are valid
* Start a call

### CODE REVIEW

File: AzureCalling/AppDelegate.swift [hunk 1]
```diff
@@ -22,17 +22,6 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func application(_ application: UIApplication, supportedInterfaceOrientationsFor window: UIWindow?) -> UIInterfaceOrientationMask {
-
```
Comment (AmyL219) [inline AzureCalling/AppDelegate.swift:25]: Could you confirm with Alex that if our sample app only supports portrait mode?

**Review by AmyL219 — APPROVED** [review]

File: AzureCalling/External/Calling/CallingContext.swift [hunk 2]
```diff
@@ -63,537 +30,34 @@ class CallingContext: NSObject {
...
-        })
-    }
-
-    private func setupCalling(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        callClient = CallClient()
-        self.setupDeviceManager(completionHandler: completionHandler)
-    }
-
-    private func setupCallAgent(displayName: String, completionHandler: @escaping (Result<Void, Error>) -> Void) {
+    func getTokenCredential() -> CommunicationTokenCredential? {
```
Comment (JoshuaLai) [inline AzureCalling/External/Calling/CallingContext.swift:33]: @AmyL219 I believe when we were chatting with CallingSDK team they recommended us not pass in `nil` as token. Should we fix this for the hero sample?

Comment (AmyL219) [reply AzureCalling/External/Calling/CallingContext.swift:33]: Good point. What CallingSDK team said is we should fetch token first. I can schedule a meeeting regarding this

Comment (JoshuaLai) [reply AzureCalling/External/Calling/CallingContext.swift:33]: Lets not block the PR cause the rest of the changes are good!

**Review by JoshuaLai — APPROVED** [review]

File: AzureCalling/AppDelegate.swift (hunk 1 above)
Comment (Leah-Xia-Microsoft) [reply AzureCalling/AppDelegate.swift:25]: > Could you confirm with Alex that if our sample app only supports portrait mode?

Good point. I'll check with Alex and per discussion, we won't block this PR to wait for a reply. We'll create a seperate PR for supporting landscape if needed.

**Pull request merged**